### PR TITLE
chore(api): Add infrastructure for a br2-external tree for buildroot

### LINKED ISSUE #2877: Balena removal: opentrons integration: Execute integration
Per #2876 integrate the buildroot and opentrons repos:

1. As an Opentrons API developer, I want to be able to do my work without ever thinking about buildroot, with as little change from the current state of affairs as possible
2. As an Opentrons system developer, I want to be able to easily coordinate changes in the API server and buildroot layers
3. As an Opentrons devops/infrastructure person, I want to always have buildroot builds available for canonically "latest" builds

These lead to this implementation strategy:

The opentrons api (and possibly the update server / some new embodiment of the update server) are added to buildroot as python packages, built from source.

The version of the api used is determined by the following rules:

* If there exists a branch of https://github.com/opentrons/opentrons that matches the current branch name, that version is used
* Otherwise, `edge` is used

On the CI side, we have the following triggers:

**On push and pr open to https://github.com/opentrons/buildroot:**

* Build buildroot using the above rules

**On push and pr open to https://github.com/opentrons/opentrons:**

* If the branch matches a branch in buildroot start a buildroot build of the matching branch (using above logic, but without the default to `edge`)
* If the branch is `edge` [optional: and the diff since last build contains changes to `api/`, `shared-data/`, `update-server/`], build `opentrons-develop` of buildroot
* If neither of those rules match, do not build buildroot

**Change `make push-api` to:**

* remount the rootfs rw
* transfer the wheel
* `pip install` the wheel
* reboot the system

(the push-api part should probably be a different target so as not to interfere with current workflows)

### PR DESCRIPTION
**Pull request by sfoster1**
More information can be found in https://buildroot.org/downloads/manual/manual.html. This change doesn't alter the normal build infrastructure; it's only used when included by buildroot externally.

This tooling provides definitions for our api server and a unit file to bring it up using systemd.

Closes #2877

### CODE REVIEW

File: external.mk [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Makefile inclusions for buildroot integration
+include $(BR2_EXTERNAL_OPENTRONS_MONOREPO_PATH)/api/buildroot.mk
+# include $(BR2_EXTERNAL_OPENTRONS_MONOREPO_PATH)/update-server/buildroot.mk
```
Comment (Laura-Danielle) [inline external.mk]: Is this comment meant to be here?

Comment (sfoster1) [reply external.mk]: I could delete it if you want, it's mostly a memory aid for when we add the update server

**Comment by sfoster1** [comment]
This is ready for review. It only works in conjunction with https://github.com/Opentrons/buildroot/pull/11/files, see that PR for more on testing

File: external.mk (hunk 1 above)
Comment (Laura-Danielle) [reply external.mk]: Nope you can keep it in if it's useful for relevant work later on.

**Pull request merged**